Sofa/Component/MechanicalLoad/src/sofa/component/mechanicalload/NodalLinearDampingForceField.inl

```diff
@@ -0,0 +1,358 @@
+                    return sofa::core::objectmodel::ComponentState::Invalid;
+                }
+            }
+        }
+    }
+
+    msg_info() << "Update from dampingCoefficientVector successfully done";
+    isConstantIsotropic = false;
+    return sofa::core::objectmodel::ComponentState::Valid;
+}
+
+template <class DataTypes>
+sofa::core::objectmodel::ComponentState
+NodalLinearDampingForceField<DataTypes>::updateFromSingleDampingCoefficient()
+{
+    // Check if the given dampingCoefficient
+    Real dampingCoefficient = d_dampingCoefficient.getValue();
+
+    if (dampingCoefficient < 0.)
+    {
+        msg_error() << "Negative \'dampingCoefficient\' given";
+        return sofa::core::objectmodel::ComponentState::Invalid;
+    }
+
+    isConstantIsotropic = true;
+    msg_info() << "Update from dampingCoefficient successfully done";
+    return sofa::core::objectmodel::ComponentState::Valid;
+}
+
+template <class DataTypes>
+void NodalLinearDampingForceField<DataTypes>::init()
+{
+    Inherit::init();
+
+    // Case no input is given
+    if (!d_dampingCoefficient.isSet() && !d_dampingCoefficientVector.isSet())
+    {
+        msg_error() << "One damping coefficient should be specified (either "
+                       "\'dampingCoefficientVector\' or \'dampingCoefficient\')";
+        this->d_componentState.setValue(sofa::core::objectmodel::ComponentState::Invalid);
+        return;
+    }
+
+    // Too many input damping information
+    if (d_dampingCoefficient.isSet() && d_dampingCoefficientVector.isSet())
+    {
+        msg_warning() << "Too many input damping information: either using \'dampingCoefficientVector\' or \'dampingCoefficient\' should be specified \ndampingCoefficient will be used";
+    }
+
+    this->d_componentState.setValue(sofa::core::objectmodel::ComponentState::Valid);
+}
+
+template <class DataTypes>
+void NodalLinearDampingForceField<DataTypes>::addForce(const core::MechanicalParams*,
+                                                       DataVecDeriv& _f, const DataVecCoord& _x,
+                                                       const DataVecDeriv& _v)
+{
+    if (!this->isComponentStateValid()) return;
+
+    SOFA_UNUSED(_x);
+
+    auto f = sofa::helper::getWriteAccessor(_f);
+    const VecDeriv& v = _v.getValue();
+
+    // Constant and isotropic damping
+    if (isConstantIsotropic)
+    {
+        const Real singleCoefficient = d_dampingCoefficient.getValue();
+
+        std::size_t i = 0;
+        for (auto& fi : f)
+        {
+            fi -= v[i++] * singleCoefficient;
+        }
+    }
+    else
+    {
+        const auto coefs = sofa::helper::getReadAccessor(d_dampingCoefficientVector);
+
+        if (!coefs.empty())
+        {
+            const std::size_t nbDampingCoeff = coefs.size();
+            std::size_t i = 0;
+
+            for (auto& fi : f)
+            {
+                const auto coefId = std::min(i, nbDampingCoeff - 1);
```

Comment:
If this code handles the case when coefs.size() != mstate.size()  by repeating the last coefficient value isn't it kind of duplicating the behavior implemented  differently between line 79-99.